packages/api-gateway/src/services/validators.ts

```diff
@@ -71,17 +83,16 @@ export function assertAvailableDescriptorExists(
   }
 }
 
-export function assertNonDraftDescriptor(
+export function assertNotValidDescriptor(
   descriptor: catalogApi.EServiceDescriptor,
-  descriptorId: catalogApi.EServiceDescriptor["id"],
   eserviceId: catalogApi.EService["id"],
   logger: Logger
-): asserts descriptor is NonDraftCatalogApiDescriptor {
-  if (descriptor.state === catalogApi.EServiceDescriptorState.Values.DRAFT) {
+): asserts descriptor is validCatalogApiDescriptor {
+  if (invalidDescriptorStates.includes(descriptor.state)) {
```

Comment:
Proposal:
I would consider exhaustive pattern matching on the state here so that if we add another state in the future, TS forces us to handle it everywhere and we don't end up again in a situation like this one, where we need to manually search all points where a state check is performed in the codebase.
I know it's more verbose, but it would help future developments like this one

Reply:
See for example what's done in `assertInterfaceDeletableDescriptorState` in the catalog-process https://vscode.dev/github/pagopa/interop-be-monorepo/blob/feature/incaricato/packages/catalog-process/src/services/validators.ts#L164